crates/primitives/src/transaction/pooled.rs

```diff
@@ -171,8 +316,28 @@ impl Decodable for PooledTransactionsElement {
 impl From<TransactionSigned> for PooledTransactionsElement {
     /// Converts from a [TransactionSigned] to a [PooledTransactionsElement].
     ///
-    /// NOTE: This will always return a [PooledTransactionsElement::Transaction] variant.
+    /// NOTE: For EIP-4844 transactions, this will return an empty sidecar.
     fn from(tx: TransactionSigned) -> Self {
-        Self::Transaction(tx)
+        let TransactionSigned { transaction, signature, hash } = tx;
```

Comment:
is an empty sidecar valid?

if so we should rethink this conversion or put the sidecare in an enum

Reply:
An empty sidecar is only valid for 4844 transactions with no versioned hashes, but we should eventually remove this conversion, right now it's used as a placeholder until we can fetch blobs for 4844 txs from the blobstore:
https://github.com/paradigmxyz/reth/blob/ca99ee2ec95cf30f3b56a700765d0cffd1ad5753/crates/net/network/src/transactions.rs#L195-L197

We can implement a method that takes a `TransactionSigned` and `Option<BlobTransactionSidecar>` to do the proper conversion